lib/ansible/modules/cloud/amazon/ec2_eip.py

```diff
@@ -159,300 +208,406 @@
+        if "Name" in tags:
+            result["name"] = tags["Name"]
 
+    return result
 
-def ensure_absent(ec2, domain, address, device_id, check_mode, isinstance=True):
+
+def ensure_absent(ec2, domain, address, device_id, release_on_disassociation, check_mode):
     if not address:
         return {'changed': False}
 
-    # disassociating address from instance
+    changed_dis = False
+    changed_rel = False
     if device_id:
-        if isinstance:
-            return disassociate_ip_and_device(ec2, address, device_id,
-                                              check_mode)
-        else:
-            return disassociate_ip_and_device(ec2, address, device_id,
-                                              check_mode, isinstance=False)
-    # releasing address
+        # disassociating address from instance
+        changed_dis = disassociate_ip_and_device(ec2, address, device_id, check_mode)
+
+    if release_on_disassociation or not device_id:
+        # releasing address
+        changed_rel = release_address(ec2, address, domain, check_mode)
+
+    if changed_dis or changed_rel:
+        result = {"changed": True}
     else:
-        return release_address(ec2, address, check_mode)
+        result = {"changed": False}
+
+    result["disassociated"] = changed_dis
+    result["released"] = changed_rel
+
+    return result
 
 
 def main():
     argument_spec = ec2_argument_spec()
     argument_spec.update(dict(
-        device_id=dict(required=False, aliases=['instance_id']),
+        device_id=dict(required=False, default=None, aliases=['instance_id']),
         public_ip=dict(required=False, aliases=['ip']),
         state=dict(required=False, default='present',
                    choices=['present', 'absent']),
-        in_vpc=dict(required=False, type='bool', default=False),
+        ec2_classic=dict(required=False, type='bool', default=False, aliases=['in_vpc']),
         reuse_existing_ip_allowed=dict(required=False, type='bool',
                                        default=False),
         release_on_disassociation=dict(required=False, type='bool', default=False),
         allow_reassociation=dict(type='bool', default=False),
-        wait_timeout=dict(default=300, type='int'),
-        private_ip_address=dict(required=False, default=None, type='str')
+        wait_timeout=dict(default=300),
+        private_ip_address=dict(required=False, default=None, type='str'),
+        name=dict(required=False, default=None, type='str'),
+        purge_tags=dict(required=False, default=False, type='bool'),
+        tags=dict(required=False, default={}, type='dict')
     ))
 
-    module = AnsibleModule(
+    module = AnsibleAWSModule(
         argument_spec=argument_spec,
         supports_check_mode=True,
         required_together=[
             ['device_id', 'private_ip_address'],
         ],
     )
 
-    if not HAS_BOTO:
-        module.fail_json(msg='boto required for this module')
+    warnings = []
+
+    if not HAS_BOTO3:
```

Comment:
no need. dependency check done by AnsibleAWSModule

Reply:
Ok.  Removed.  Thanks for the review.